### ISSUE TITLE
Error when using ignoreTemplateLiterals for max-len

### ISSUE THREAD

**Issue by CorentinAndre** [eslint/eslint#9801]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 8.9.0
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

// https://eslint.org/docs/user-guide/configuring

module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint',
    sourceType: 'module'
  },
  env: {
    browser: true,
  },
  extends: ['airbnb-base', 'plugin:vue/essential'],
  // required to lint *.vue files
  plugins: [
    'html'
  ],
  // check if imports actually resolve
  settings: {
    'import/resolver': {
      webpack: {
        config: 'build/webpack.base.conf.js'
      },
      "babel-module": {}
    }
  },
  // add your custom rules here
  rules: {
    // don't require .vue extension when importing
    'import/extensions': ['error', 'always', {
      js: 'never',
      vue: 'never'
    }],
    'max-len': ['error', {
      code: 100,
      ignoreTemplateLiterals: true,
    }],
    // allow optionalDependencies
    'import/no-extraneous-dependencies': ['error', {
      optionalDependencies: ['test/unit/index.js']
    }],
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off'
  }
}


```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js


<template>
  <div id="app">
    <notifications/>
    <router-view class="container"/>
  </div>
</template>

<script>
const test = 12;
export default {
  name: `WOW this is a very long title using template literals  wow wow wow wow wow wow  wow wow wow wow wow ${test}`,
};
</script>

<style lang="scss">
@font-face {
  font-family: 'Avenir Next';
  src: url('~@/assets/fonts/AvenirNext-Bold.ttf') format('truetype');
  font-weight: $bold;
  font-style: normal;
}
@font-face {
  font-family: 'Avenir Next';
  src: url('~@/assets/fonts/AvenirNext-Regular.ttf') format('truetype');
  font-weight: $regular;
  font-style: normal;
}
@font-face {
  font-family: 'Avenir Next';
  src: url('~@/assets/fonts/AvenirNext-Italic.ttf') format('truetype');
  font-weight: $regular;
  font-style: italic;
}
@font-face {
  font-family: 'Avenir Next Demi';
  src: url('~@/assets/fonts/AvenirNext-DemiBold.ttf') format('truetype');
  font-weight: $demi-bold;
  font-style: normal;
}
@font-face {
  font-family: 'Avenir Next';
  src: url('~@/assets/fonts/AvenirNext-Medium.ttf') format('truetype');
  font-weight: $medium;
  font-style: normal;
}
@font-face {
  font-family: 'Avenir Next';
  src: url('~@/assets/fonts/AvenirNext-MediumItalic.ttf') format('truetype');
  font-weight: $medium;
  font-style: italic;
}
#app {
  font-family: -apple-system, BlinkMacSystemFont,Roboto, 'Avenir Next', sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: $regular;
  font-size: $normal;
  color: $black;
}
</style>



```

```bash
npm run dev
```

**What did you expect to happen?**
I wanted eslint to ignore template literals for max-len rule. It should ignore the line with a template literal of more than a 100 chars.
**What actually happened? Please include the actual, raw output from ESLint.**
Line is parsed and eslint throws an error.

```bash
 WAIT  Compiling...                                                                                                                                                       15:22:49

 95% emitting

 WARNING  Compiled with 1 warnings                                                                                                                                        15:22:50


  ✘  http://eslint.org/docs/rules/max-len  Line 3 exceeds the maximum line length of 100
  src/App.vue:11:1
    name: `WOW this is a very long title using template literals  wow wow wow wow wow wow  wow wow wow wow wow ${test}`,
   ^
```

**Comment by mysticatea** [comment]
Thank you for the report.

However, I could not reproduce it with the latest versions: [Online demo](https://mysticatea.github.io/vue-eslint-demo/#eJzFVW1L3EAQ/ivDVjiVpNF+zJ1FbREKtQUV/NAruJfMxcHNbtid3IvH9bd3NjmtpRZ60FZIwu5kZp5nnh1mV6pwJapcjRjrxmjGt2MLMCppBlQejZVumrHqbGK1jmlKhWZyNmQPVu9aRp/OCOdQGB2ChBXOsiaLfqx6v1EmKWU1yp4AyTYUnhqWdbaPwZBlqPUiNWhz+ILeO5/ACiLHHA4PDhKgyjqPV5scH0mQtQk5sG8R1l9hP4tpBT4wMMrnCA7fDMcWF43zDCVOdWsYVpGT1bWkvbn+fA18SwHk0TBDvwTjbAVMbBDaQHG9AQSzQQSYu/mz77PGnVUks75JxnY9jCo81h014KUAGW0rUS4UIXSKH09Fw3SqC+zZbrY1mWUOg5MZWvLwCRc8GMbfwRc5tN7sDr4dZ3IIyCGLISHrXaNneupM+Zp5OtiTdL7WvDuIwvGywcHe8BFljlTdcg47E/H/Ye545mBjpBHz+t+QvMCqNdpvw9P3If+d6gfWhoq/wZS6TFsyhfdY0xZ0o/u2LVBKTPoifXCOJbX1NlzrLuKFiG7fC7+h+7QVXsn8fYZoKmaDaVgGGUwJnMrgvDvXxWW3PxPP5MJNHLvk55ISCNqGNKCnaQcrXCZ3JHk7+No5mYO2ykFbJiGhA/bHntbuPnVh8Ytj5fUyFNrgH7U63Ut5O4/HATLYjfNxzhhd3PUVy2iMMsgAVInyrcGg8tU6UWRLtHwpKVT+5mF7JdLK5RUaOU9xFz3Z+Y0xXhontjyjBZbv4iWXqEZ7qV3+TfQETdpfOGr9HRKXd54=).

ESLint `3.19.0` is old one. Would you try it with the latest versions?

**Comment by CorentinAndre** [comment]
Perfectly working in next version. 
Thanks a lot. Closing it since it is fixed.
